pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/QueryQuotaClusterIntegrationTest.java

```diff
@@ -125,12 +132,15 @@ public void testDatabaseQueryQuotaWithTableQueryQuotaWithExtraBroker()
       addQueryQuotaToTableConfig(10);
       // Add one more broker such that quota gets distributed equally among them
       brokerStarter = startOneBroker(2);
+      _brokerBaseApiUrl = LOCAL_HOST + ":" + brokerStarter.getPort();
       // to allow change propagation to QueryQuotaManager
       Thread.sleep(1000);
```

Comment:
Sleep is not robust, especially in the CI environment where machines have very limited resource. One way to make it more robust is to use `TestUtils.waitForCondition()` and keep trying until it meets the expectation.